### ISSUE TITLE
Add support for the new cargo lockfile format

### ISSUE THREAD

**Issue by jonasbb** [xd009642/tarpaulin#299]
Recently cargo nightly started produces lockfiles with the new format (https://github.com/rust-lang/cargo/pull/7579).

Tarpaulin fails to compile tests, if it encounters such a log file. One instance can be found [here](https://travis-ci.org/jonasbb/serde_with/jobs/618642545). Tarpaulin fails with this message:
```
Error: "Failed to compile tests! Error: failed to parse lock file at: /home/travis/build/jonasbb/serde_with/Cargo.lock"
```

In my case I run Tarpaulin on a library and therefore do not commit the lockfile, which is why nightly now uses the new lockfile format.

**Comment by xd009642** [comment]
There's now a new cargo_update branch this updates the tarpaulin cargo dependency to 0.40.0 and may solve the issue. I'll try it myself when I've got time to but in the meantime if you test it and report back I can fast track the merge/release :+1:

**Comment by HaoXuan40404** [comment]
Same problem too

**Comment by jonasbb** [comment]
@xd009642 I started a try build with tarpaulin from the new branch. It passed on all Rust versions.
https://travis-ci.org/jonasbb/serde_with/builds/621365163
So this would fix the issue for me.

**Comment by xd009642** [comment]
Awesome, well since it doesn't affect anything in terms of existing build functionality I've just pushed out another release. If you could try it out and confirm it now works this issue can be closed :smile:

**Comment by xd009642** [comment]
It's now released in 0.10.0 so closing this
